lib/spack/spack/filesystem_view.py

```diff
@@ -413,7 +413,24 @@ def remove_specs(self, *specs, **kwargs):
         remove_extension = ft.partial(self.remove_extension,
```

Comment:
Since you have introduced sorting, the comment right above this line can now be removed: could you do that?

Reply:
Done, and re-reading this comment made me think that the sorting should be applied to all packages to be removed (rather than just extensions). So this is done in the update, which also simplifies the surroundings.